### PR TITLE
Adds middleware to restrict access of unauthorised user

### PR DESCRIPTION
**Pull request by rehan99000**
**Description:**
Adds a middleware that gives 404 if requesting user does not have access to current site. Adds test

**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-1323

### CODE REVIEW

File: ecommerce/extensions/edly_ecommerce_app/tests/test_middlewares.py [hunk 1]
```diff
@@ -76,3 +77,72 @@ def test_settings_override_middleware_overrides_settings_correctly(self):
...
+        Create environment for Edly organization access middleware tests.
+        """
+        super(EdlyOrganizationAccessMiddlewareTests, self).setUp()
+        self.dashboard_url = reverse('dashboard:index')
+        self.user = self.create_user()
+        self.client.login(username=self.user.username, password='test')
+        self.test_edly_user_info_cookie_data = {
+            'edly-org': 'edly',
+            'edly-sub-org': 'cloud',
+            'edx-org': 'edx'
```
Comment (zubair-arbi) [inline ecommerce/extensions/edly_ecommerce_app/tests/test_middlewares.py:97]: nit: add comma `,` at the end of last item in python list or dict.

Comment (rehan99000) [reply ecommerce/extensions/edly_ecommerce_app/tests/test_middlewares.py]: done

File: ecommerce/extensions/edly_ecommerce_app/helpers.py [hunk 2]
```diff
@@ -71,3 +71,20 @@ def is_valid_site_course(course_id, request):
         return True
 
     return False
+
+def user_has_edly_organization_access(request):
```
[the remark this answers is not in the document]
Comment (rehan99000) [reply ecommerce/extensions/edly_ecommerce_app/helpers.py:75]: done

[the remark this answers is not in the document]
Comment (zubair-arbi) [reply ecommerce/extensions/edly_ecommerce_app/helpers.py:75]: Still need to address this.

[the remark this answers is not in the document]
Comment (rehan99000) [reply ecommerce/extensions/edly_ecommerce_app/helpers.py:75]: done

**Pull request merged**